Backpack/Price/Classes/BPKPrice.swift

```diff
@@ -154,30 +136,68 @@ public final class BPKPrice: UIView {
                 priceStackView.spacing = BPKSpacingNone
             }
         }
-        
-        containerStackView.addSubview(priceStackView)
-        addSubview(containerStackView)
     }
     
-    private func setupConstraints() {
-        NSLayoutConstraint.activate([
-            containerStackView.topAnchor.constraint(equalTo: topAnchor),
-            containerStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
-            containerStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
-            containerStackView.bottomAnchor.constraint(equalTo: bottomAnchor)
-        ])
-    }
-    
-    private func footnoteOrCaptionFontStyle() -> BPKFontStyle {
+    private func accessoryFontStyle() -> BPKFontStyle {
         return size == .large ? .textFootnote : .textCaption
     }
     
     private func applyLineThroughStyling() {
-        guard let lineThroughText = lineThroughText else {
-            lineThroughTextLabel.attributedText = nil
+        guard let previousPrice = previousPrice else {
+            previousPriceLabel.attributedText = nil
             return
         }
-        let attributedString = NSAttributedString(string: lineThroughText, attributes: strikeThroughTextAttributes)
-        lineThroughTextLabel.attributedText = attributedString
+        let attributedString = NSAttributedString(string: previousPrice, attributes: strikeThroughTextAttributes())
+        previousPriceLabel.attributedText = attributedString
+    }
+    
+    private func stylePriceLabel() {
+        priceLabel.fontStyle = size == .large ? .textHeading2 : .textHeading4
+        priceLabel.textColor = BPKColor.textPrimaryColor
+    }
+    
+    private func styleAccessoryLabels() {
+        [
+            trailingTextLabel,
+            previousPriceLabel,
+            separatorLabel,
+            leadingTextLabel
+        ].forEach {
+            $0.fontStyle = accessoryFontStyle()
+            $0.textColor = BPKColor.textSecondaryColor
+        }
+    }
+    
+    private func styleTopTextStackView() {
+        [previousPriceLabel, separatorLabel, leadingTextLabel].forEach {
+            topTextStackView.addArrangedSubview($0)
+        }
+        topTextStackView.axis = .horizontal
+        topTextStackView.spacing = BPKSpacingSm
+    }
+    
+    private func stylePriceStackView() {
+        [priceLabel, trailingTextLabel].forEach {
+            priceStackView.addArrangedSubview($0)
+        }
+        priceStackView.axis = .horizontal
+        priceStackView.spacing = BPKSpacingSm
+        priceStackView.alignment = .firstBaseline
+    }
+    
+    private func styleContainerStackView() {
+        [topTextStackView, priceStackView].forEach {
+            containerStackView.addArrangedSubview($0)
+        }
```

Comment:
having this logic inside the style method can be a bit confusing and hard to follow code in the future, please keep the view setup logic in the setupView method.